### PR TITLE
WIP Move conferences CRUD to Orga namespace

### PR DESCRIPTION
**Pull request by F3PiX**
#463

Steps:
- [x]  Add conferences_controller to orga namespace
- [x] Remove CRUD actions from conferences_controller
- [x] Add _table partial for shared views
- [x] Render table from both index views
- [x] Restore sortable columns in table (translate from slim)
- [x] show
- [x] other views
- [x] Check redirects from orga/conferences to conferences
- [x] Check abilities and authorisations
- [x] What to do with routing tests? Destroy :trollface: 

Other todo's:
- [x] Translate first round of conferences from slim to erb and add toggle  
- [x] small issues from #503
  - [x] redirect after failing validation
  - [x] restrict numbers in selection fields
- [x] Add tests for authorisations???
- [x] Make 'back to all'-link staying in relevant namespace

Left-overs
- link back from Calendar view
- restrict end dates that are before the start date #503 -2
- adjust js toggle to show the upcoming round of conferences

### CODE REVIEW

File: spec/routing/conferences_routing_spec.rb [hunk 1]
```diff
@@ -10,19 +10,19 @@
       expect(get('/conferences/1')).to route_to('conferences#show', id: '1')
     end
 
-    it 'routes to #edit' do
+    xit 'routes to #edit' do
```
Comment (F3PiX) [inline spec/routing/conferences_routing_spec.rb]: What should I do with this routing tests? I can move them to an spec/routing/orga/conferences_routing_spec.  But do we need routing_specs in the regular controller as well?

**Comment by F3PiX** [comment]
In order to finish this, I have a few questions:
1) Can I delete the routing tests? Or what do I do with them?
2) When I type `/conferences/new` url, I expect one of two things: either a message that the route doesn't exist, or a message that acces is denied. However, I get a message:  `Couldn't find conference with id=new`.  Is this okay, or did I do something wrong with either authorization or the routing?
3) Because all the CRUD is behind the orga base controller, I removed all `can? :crud, Conference`-checks in the CRUD views. okay, right?
4) I tried to [add a validation](https://github.com/rails-girls-summer-of-code/rgsoc-teams/pull/546/commits/50d659c38edb806d58e94752aac9eb80e682e891) to ensure that the end_date will be after the start_date. (issue #503 nr 2) But... it would not allow for ANY end date at all, so I removed it in a later commit. Any suggestions?

(ping @carpodaster , @michaelem , @ramonh )

File: app/views/conferences/_table.html.erb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    <td><%= conference.lightningtalkslots? ? "yes" : "no" %></td>
+  </tr>
+      <% end %>
+    </tbody>
+</table>
+
+<br>
+<p><%= link_to "View as Calendar", calendar_path %>
+</p>
+<!--TODO translate first round to erb-->
```
Comment (F3PiX) [inline app/views/conferences/_table.html.erb]: I will remove this.

File: spec/routing/conferences_routing_spec.rb (hunk 1 above)
Comment (carpodaster) [inline spec/routing/conferences_routing_spec.rb]: Personally, I don't write routing spec for "normal", ie. restful controllers. I find them to be useful when you do something fancy with parameter parsing, routing contraints, or generally diverging from the expected pathname for a given controller.

The conferences controller is pretty standard. I'd have no problem with deleting these examples 😱

**Comment by carpodaster** [comment]
Hej Maud!

re: 2) You are right, you should get a routing error. Can you verify that there is no `new.html` template in the original view folder? I see it was deleted from the repo in your PR, but maybe it's still hanging around on your computer?

re: 3) Yes, simplifying the authorization is one of the premier goals of separating all the things into different namespaces. I'll shoot from the hip and say: yep, `can? :foo` checks (along with the definitions in `ability.rb` begone! 🔫 

re: 4) Let's concentrate on just moving the files into the orga namespace in this PR. Always be wearing [just one hat at a time](http://moleseyhill.com/blog/2009/05/18/kent-becks-four-hats/).

**Comment by F3PiX** [comment]
Thanks Carsten! 
re: 2) Computer says: no `new.html` in `conferences`. The funny thing is, the `/new`-part is read as if it is a param, so the `show`-view is called.  I think now that it has to do with the conference helper method (here)[https://github.com/rails-girls-summer-of-code/rgsoc-teams/pull/546/files#diff-7fbe00a3a9ebba1c66f31e2057134f9aR12], but I did not find a way to solve it. 

re:3) Missed a view things ;-)

re: 4) I am going to call you LinkMeister from now on.  And I will watch my hats. Thnx

**Comment by F3PiX** [comment]
re: 1) I found that the `/users/new`-route behaves exactly the same. => 404 instead of a neat and informative redirect. (TIL: load_and_authorize_resource doesn't work on non-existing routes)
One solution I came up with: restore all `conferences` routes, and add a redirect_to orga_conferences-path in the actions. So, the orga::base_controller takes care of access rights and messages. Wdyt, @carpodaster?

**Comment by F3PiX** [comment]
@carpodaster I thought this was about ready. What do you think?

**Comment by carpodaster** [comment]
@F3PiX thank you! I have a busy weekend ahead but will look into it next Monday

**Comment by carpodaster** [comment]
Thanks a lot @F3PiX - looks good! Let's merge this one :smile:

**Comment by F3PiX** [comment]
:tada: :tulip:

**Pull request merged**